README.md

```diff
@@ -306,49 +649,92 @@ dojo --docker-options="--init" --image=alpine:3.8 -i=false sh -c "echo 'will sle
 
-Alternatively, you can build the binary file yourself, see [Development](#development).
+Still vagrant is a great tool and irreplaceable if you must use a VM.
 
-## Aims
-Dojo is intended to be used with **short-running commands**, e.g. to compile code, to create backup dump, to use a 3rd party software to do some operation like e.g. use [Terraform](https://www.terraform.io/) to create a VM or run some graphical IDE for local development. It is rather not expected to use Dojo with long-running commands, e.g. to run a server 24/7.
+`Dojofile` is similar to `Vagrantfile` in its function. But with a restriction that Dojo forces you to first build and release the image, while `Vagrantfile` has some tools to provision the VM when running `vagrant up` which adds up to the startup time.
 
-Dojo was invented in order to resolve the problems of:
-   * **fast changing work environment requirements** (you had to reinstall/reconfigure over and over again)
-   * using **different environments at the same moment** (e.g using Java 7 for some projects and Java 8 for other)
-   * **work environments becoming bloated** (in one machine you have to have installed and configured: Mono, Java, Python, etc.)
-   * **not reproducible work environments** - **works on my machine** (e.g. you installed Mono just fine on your work computer, but then installing it on your laptop demanded some additional steps or resulted in having installed different versions). This is especially important if you use  **Continuous Integration**, the environment should be the same on workstation and CI agents.
-   * **mutable work environments** - e.g. your code compiled a month ago, but you installed sth new and it influenced the environment
-   * **experimenting with new technologies without breaking your workstation**
+### vs. dependency managers
 
-## Limitations
-Dojo was tested only on Linux, with local Docker server.
+A typical dependency manager allows you to commit references to all immediate and transitive dependencies in source control, files such as `Gemfile.lock`, `paket.lock`, `yarn.lock` etc. It is a good practice to do so, because project defines fully, in code, what libraries it depends on.
+If we extend this concept, then we soon realize that the project did not define how the host, where it should be built, should look like. Probably most common situation is that there is a README file listing several tools to install. It is also the least helpful situation.
+In any sensible organization, configuration management is used to provision the CI-agents and developer's workstation building the project. It is one level better than the README, but still the definition of the environment exists outside the project and there is no reference to it.
+The `Dojofile` is the *lock* file which allows to store this reference in source control. When you run `dojo <command>`, then we first fetch the environment, just like a dependency manager would fetch all dependencies of the project.
```

Comment:
Place replace: "When you run `dojo <command>`, then we first fetch the environment" with "When you run `dojo <command>`, then the environment is fetched".